src/foam/u2/Element.js

```diff
@@ -110,9 +110,13 @@ foam.CLASS({
           foam.__context__;
 
         // Install our own CSS, and then all parent models as well.
-        if ( X.document && ! axiom.installedDocuments_.has(X.document) ) {
-          X.installCSS(axiom.expandCSS(this, axiom.code), cls.id);
-          axiom.installedDocuments_.set(X.document, true);
+        if ( X.document ) {
+          var map = axiom.installedDocuments_.get(X.document) || {};
+          if ( ! map[this.id] ) {
+            X.installCSS(axiom.expandCSS(this, axiom.code), this.id);
```

Comment:
I noticed you changed the second argument on this line from `cls.id` to `this.id`. Was that intentional?

Reply:
Yeah it's intentional. `this` will be the class that's being created, which isn't necessarily `cls` when dealing with inheritance.